src/MessageTrait.php

```diff
@@ -66,7 +66,7 @@ public function getProtocolVersion()
      * new protocol version.
      *
      * @param string $version HTTP protocol version
-     * @return self
+     * @return static
```

Comment:
@Ocramius No, it does not. The interface indicates uses `@return self`, and indicates a new instance is returned. Considering that `MessageInterface` is not expected to be directly implemented, but rather implemented via one of `RequestInterface` or `ResponseInterface`, the question is whether `self` or `static` should be specified in the return annotation to ensure that IDEs, PHPDocumentor, and implementors are following the spec properly.